File: Python/744. Find Smallest Letter Greater Than Target.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
#         letters.append(target)
#         letters.sort(reverse = True)
#         temp = letters.index(target) - 1
#         if temp < 0:
#             return letters[-1]
#         else:
#             return letters[temp]

        for i in letters:
            if i > target: return i
        return letters[0]

File: C/744. Find Smallest Letter Greater Than Target.c
char nextGreatestLetter(char* letters, int lettersSize, char target){
    int t = (int)target;

    // for (int i = 0; i < lettersSize; i++) {
    //     if ((int)letters[i] > t) return letters[i];
    // }

    int l = 0, r = lettersSize - 1, m;

    while (l <= r) {
        m = (l + r) / 2;

        if ((int)letters[m] > t) r = m - 1;
        else l = m + 1;
    }

    if (l >= lettersSize) return letters[0];
    else return letters[l];
}
